lib/torrent.js

```diff
@@ -1302,35 +1306,38 @@ export default class Torrent extends EventEmitter {
    * Garbage collect selections with respect to the store's current state.
    */
   _gcSelections () {
-    for (let i = 0; i < this._selections.length; ++i) {
-      const s = this._selections[i]
+    for (const s of this._selections) {
       const oldOffset = s.offset
 
       // check for newly downloaded pieces in selection
       while (this.bitfield.get(s.from + s.offset) && s.from + s.offset < s.to) {
         s.offset += 1
       }
 
-      if (oldOffset !== s.offset) s.notify()
+      if (oldOffset !== s.offset) {
+        this._selections.getNotificationsByRange(s.from, s.from + s.offset).forEach(({ notify }) => {
```

Comment:
this... is gonna be insanely slow when there are a lot of selections?, also use for of, not forEach